### DISCUSSION TITLE
there is a strange AssertionError

### DISCUSSION THREAD

**Discussion by EEEEEREN** [STVIR/pysot#527] [Q&A] [answered]
when i download the pysot from github
and set the environment properly, i can run demo

but when i try to test , there is a strange error , like this
![QQ截图20210328014631](https://user-images.githubusercontent.com/67900018/112729516-6f801300-8f67-11eb-83bf-a5cd9f394792.png)
i have already put the testing dataset  VOT2018 in a correct path as you see
![image](https://user-images.githubusercontent.com/67900018/112729542-9e968480-8f67-11eb-8b20-6218650f07b0.png)

but this Error shows a strange path,  i didnt put any dataset in siamrpn_r50_l234_dwxcorr 
==============================================================
报了一个很奇怪的错误，做测试的时候，显示这个路径下的AssertionError，但我肯定没有把数据集放这里，是按照正确的位置放的数据集，数据集里面也是正确无误的。但是做测试的时候，发现路径根本不对，报错的路径根本没有数据集，却报这样的错误。
不知道什么时候开始，从github上下下来的代码，就会有这样的问题，我测试了好几台机器了，跑我之前下的旧代码就没问题。我肯定是按照readme上面一步一步来，跑完demo之后接下来测试就报这个奇怪的错。

**Comment by ZhiyuanChen** [comment, marked as answer]
You probably want to run the code in `~/test/pysot/experiments/`

**Comment by empty985** [comment]
你弄好了吗？我也报这个错误了，不知道怎么回事

**Comment by EEEEEREN** [comment]
> 你弄好了吗？我也报这个错误了，不知道怎么回事

我直接删掉了新下载的代码，用的别人的pysot代码

**Reply by Xushibo96** [reply]
`toolkit/datasets/video.py `文件中 18行  
`self.img_names = [os.path.join(os.path.abspath(root), os.path.abspath(x)) for x in img_names] ` 
里面的`os.path.abspath(x)` 改为 `x`就可以了
